test/integration/simple_operations_test.lua

```diff
@@ -1007,3 +1007,139 @@ pgroup:add('test_partial_result_bad_input', function(g)
     t.assert_str_contains(err.err, 'Space format doesn\'t contain field named "lastname"')
 end)
 
+pgroup:add('test_opts_not_damaged', function(g)
```

Comment:
I have reverted changes for CRUD and etst failed on asserts that checks opts. I have commented out these checks:
```lua
--- a/test/integration/simple_operations_test.lua
+++ b/test/integration/simple_operations_test.lua
@@ -1036,7 +1036,7 @@ pgroup:add('test_opts_not_damaged', function(g)
     ]], {insert_opts})
 
     t.assert_equals(err, nil)
-    t.assert_equals(new_insert_opts, insert_opts)
+    --t.assert_equals(new_insert_opts, insert_opts)
 
     -- upsert
     local upsert_opts = {timeout = 1, bucket_id = 907, fields = {'name', 'age'}}
@@ -1068,7 +1068,7 @@ pgroup:add('test_opts_not_damaged', function(g)
     ]], {upsert_opts})
 
     t.assert_equals(err, nil)
-    t.assert_equals(new_upsert_opts, upsert_opts)
+    --t.assert_equals(new_upsert_opts, upsert_opts)
 
     -- get
     local get_opts = {timeout = 1, bucket_id = 401, fields = {'name', 'age'}}
@@ -1128,7 +1128,7 @@ pgroup:add('test_opts_not_damaged', function(g)
     ]], {replace_opts})
 
     t.assert_equals(err, nil)
-    t.assert_equals(new_replace_opts, replace_opts)
+    --t.assert_equals(new_replace_opts, replace_opts)
 
     -- delete
     local delete_opts = {timeout = 1, bucket_id = 401, fields = {'name', 'age'}}
```
and test passed. Why?
I expected that test should fail with an error described in #192: "unexpected argument opts.add_space_schema_hash to get".

Reply:
This error will occur if damaged options (ex. `upsert_opts`) to the other method (ex. `get`) that does not accept `add_space_schema_hash` as an opts table field. In this test I don't pass opts table from one method to the other, I am just checking that opts table was not modified by each of CRUD method.

Reply:
I would reproduce the original problem in a test and then fix it in a code. Right now (and in the future) we cannot guarantee that problem has been fixed.

Reply:
But the problem was in modification `opts` parameter inside `crud` methods and tests check that passed `opts` parameter and received `opts` parameter are the same.